### ISSUE TITLE
Update script_runner.mojo docstring example after signature change

### ISSUE THREAD

**Issue by mvillmow** [HomericIntelligence/ProjectOdyssey#3881]
The module-level docstring example in `shared/training/script_runner.mojo` still shows the old API (`run_epoch_with_batches` without `step_fn`). It should be updated to show a working call with a real `DataLoader` and a `step_fn` argument so users aren't misled by the stale example.

_Follow-up from #3284_

**Comment by mvillmow** [comment]
# Implementation Plan

Now I have enough context. Here's the implementation plan:

---

## Implementation Plan: Issue #3881 — Update `script_runner.mojo` Module Docstring Example

### 1. Objective

Update the module-level docstring in `shared/training/script_runner.mojo` (lines 1–22) to replace the stale example (which omits `run_epoch_with_batches` and `step_fn`) with a working example showing `run_epoch_with_batches` called with a real `DataLoader` and a `step_fn` argument.

---

### 2. Approach

- **Minimal change**: Only update the module-level docstring example. No code changes.
- **Show the full current API**: Include `run_epoch_with_batches` with `loader`, `callbacks`, and a `step_fn` lambda/function.
- **Line-length discipline**: Keep every line of example content ≤ 70 chars to avoid `mojo format` reformatting the closing `"""`.
- **Pre-emptive multiline closing**: Place the closing `"""` on its own line since the docstring is multi-line.
- **Preserve existing imports shown**: Keep `TrainingCallbacks`, `print_training_header`, etc. since they remain valid — just add `run_epoch_with_batches` and `DataLoader`/`create_simple_dataloader` to the example.

---

### 3. Files to Create

None.

---

### 4. Files to Modify

**`/home/mvillmow/ProjectOdyssey/shared/training/script_runner.mojo`** — lines 1–22 (module docstring only)

Replace the existing example block with one that:

1. Imports `run_epoch_with_batches` (and optionally `DataLoader` via `create_simple_dataloader` from `trainer_interface`)
2. Shows constructing a `DataLoader` using `create_simple_dataloader`
3. Defines or references a `step_fn` with signature `fn (ExTensor, ExTensor) raises -> ExTensor`
4. Calls `run_epoch_with_batches(loader, callbacks, step_fn)`

**Proposed new docstring:**

```
"""Training script runner utilities.

Provides reusable components for paper training scripts including
callbacks, epoch runners, and display utilities.

Example:
    ```mojo
    from shared.training.script_runner import (
        TrainingCallbacks,
        run_epoch_with_batches,
        print_training_header,
    )
    from shared.training.trainer_interface import (
        create_simple_dataloader,
    )
    from shared.core.extensor import ExTensor

    fn step(x: ExTensor, y: ExTensor) raises -> ExTensor:
        return x  # replace with real forward+loss

    var loader = create_simple_dataloader(
        data^, labels^, batch_size=32
    )
    var callbacks = TrainingCallbacks(verbose=True)
    var loss = run_epoch_with_batches(
        loader, callbacks, step
    )
    ```
"""
```

> **Note**: Check exact character counts before committing. If any line inside the fenced block (including the 4-space indent) exceeds ~88 chars, split it. The closing `"""` must be on its own line.

---

### 5. Implementation Order

1. **Read** the exact current docstring in `shared/training/script_runner.mojo` (lines 1–22) to capture the precise `old_string` for `Edit`.
2. **Draft** the new docstring, verifying every line is ≤ 70 content characters.
3. **Edit** the file using the `Edit` tool with `old_string` = exact current docstring block, `new_string` = new docstring block.
4. **Run** `pixi run pre-commit run --all-files` to verify `mojo format`, `markdownlint`, and trailing-whitespace hooks all pass.
5. **Fix** any formatting issues surfaced by pre-commit (re-edit if `mojo format` splits a line).
6. **Commit** with message: `docs(script_runner): update module docstring example for run_epoch_with_batches`
7. **Push** and open PR linked to issue #3881.

---

### 6. Verification

- `pixi run pre-commit run --all-files` passes with no diff (especially `mojo format` and `trailing-whitespace` hooks).
- The example in the docstring compiles conceptually: imports exist, `create_simple_dataloader` signature matches `(var ExTensor, var ExTensor, Int) -> DataLoader`, `step_fn` signature matches `fn (ExTensor, ExTensor) raises -> ExTensor`.
- No other files changed (diff limited to `shared/training/script_runner.mojo` docstring block).

---

### 7. Skills Used

**Team Knowledge Base Skills (from Prior Learnings):**

| Skill | How Applied |
|-------|-------------|
| `mojo-docstring-format-precommit` | Drove line-length discipline (≤70 content chars), pre-emptive closing `"""` on own line, and the `pixi run pre-commit run --all-files` verification step |
| `mojo-note-to-docstring` | Informed docstring pattern conventions and the pre-commit verification workflow |

**Skills invoked during planning:** None directly invoked (used `Read` and `Grep` tools directly to inspect the file and current API signatures).

---
*Generated by Claude Code Planner*
